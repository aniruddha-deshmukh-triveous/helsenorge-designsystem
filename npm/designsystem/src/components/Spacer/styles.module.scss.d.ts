export type Styles = {
  spacer: string;
  'spacer--2xl': string;
  'spacer--2xs': string;
  'spacer--3xl': string;
  'spacer--3xs': string;
  'spacer--4xl': string;
  'spacer--4xs': string;
  'spacer--5xl': string;
  'spacer--6xl': string;
  'spacer--l': string;
  'spacer--m': string;
  'spacer--s': string;
  'spacer--xl': string;
  'spacer--xs': string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
