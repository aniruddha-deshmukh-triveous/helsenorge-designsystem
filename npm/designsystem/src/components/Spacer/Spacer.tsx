import React from 'react';
import StyledSpacer from './Spacer.styled';
import { SpacerSizes } from './../../theme/spacers';

interface SpacerProps {
  /** Sets the size of the spacer. */
  size?: SpacerSizes;
  /** Adds custom classes to the element. */
  className?: string;
}

const Spacer = React.forwardRef(function SpacerForwardedRef(props: SpacerProps, ref: React.ForwardedRef<HTMLElement>) {
  const { size = 's', className } = props;

  return <StyledSpacer size={size} className={className} ref={ref}></StyledSpacer>;
});

export default Spacer;
