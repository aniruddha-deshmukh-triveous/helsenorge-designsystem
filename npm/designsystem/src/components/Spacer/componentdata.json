{
  "props": {
    "size": {
      "defaultValue": null,
      "description": "Sets the size of the spacer.",
      "name": "size",
      "parent": {
        "fileName": "src/components/Spacer/Spacer.tsx",
        "name": "SpacerProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Spacer/Spacer.tsx",
          "name": "SpacerProps"
        }
      ],
      "required": false,
      "type": {
        "name": "enum",
        "raw": "\"3xs\" | \"2xs\" | \"xs\" | \"s\" | \"m\" | \"l\" | \"xl\" | \"2xl\" | \"3xl\" | \"4xl\" | \"5xl\" | \"6xl\" | undefined",
        "value": [
          {
            "value": "undefined"
          },
          {
            "value": "\"3xs\""
          },
          {
            "value": "\"2xs\""
          },
          {
            "value": "\"xs\""
          },
          {
            "value": "\"s\""
          },
          {
            "value": "\"m\""
          },
          {
            "value": "\"l\""
          },
          {
            "value": "\"xl\""
          },
          {
            "value": "\"2xl\""
          },
          {
            "value": "\"3xl\""
          },
          {
            "value": "\"4xl\""
          },
          {
            "value": "\"5xl\""
          },
          {
            "value": "\"6xl\""
          }
        ]
      }
    },
    "className": {
      "defaultValue": null,
      "description": "Adds custom classes to the element.",
      "name": "className",
      "parent": {
        "fileName": "src/components/Spacer/Spacer.tsx",
        "name": "SpacerProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Spacer/Spacer.tsx",
          "name": "SpacerProps"
        }
      ],
      "required": false,
      "type": {
        "name": "string"
      }
    }
  }
}
