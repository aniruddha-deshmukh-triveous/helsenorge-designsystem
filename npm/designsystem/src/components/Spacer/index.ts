import Spacer from './Spacer';
export * from './Spacer';
export default Spacer;
