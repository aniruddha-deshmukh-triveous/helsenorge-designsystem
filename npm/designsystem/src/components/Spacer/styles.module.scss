.spacer {
  display: block;
  width: 100%;
  height: 0.125rem;

  &--2xs {
    height: 0.25rem;
  }
  &--xs {
    height: 0.5rem;
  }
  &--s {
    height: 1rem;
  }
  &--m {
    height: 1.5rem;
  }
  &--l {
    height: 2rem;
  }
  &--xl {
    height: 3rem;
  }
  &--2xl {
    height: 4rem;
  }
  &--3xl {
    height: 5rem;
  }
  &--4xl {
    height: 6rem;
  }
  &--5xl {
    height: 7rem;
  }
  &--6xl {
    height: 8rem;
  }
}
