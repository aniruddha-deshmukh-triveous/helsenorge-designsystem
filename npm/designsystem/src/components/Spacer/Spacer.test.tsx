import React from 'react';
import { render } from '@testing-library/react';
import Spacer from './Spacer';

test('displays default spacer', (): void => {
  const { container } = render(<Spacer />);
  expect(container).toMatchSnapshot();
  expect(container.querySelector('span').className).toBe('spacer spacer--s');
});

test('displays 3xs spacer', (): void => {
  const { container } = render(<Spacer size="4xs" />);
  expect(container.querySelector('span').className).toBe('spacer spacer--4xs');
});

test('displays 3xs spacer', (): void => {
  const { container } = render(<Spacer size="3xs" />);
  expect(container.querySelector('span').className).toBe('spacer spacer--3xs');
});

test('displays 2xs spacer', (): void => {
  const { container } = render(<Spacer size="2xs" />);
  expect(container.querySelector('span').className).toBe('spacer spacer--2xs');
});

test('displays xs spacer', (): void => {
  const { container } = render(<Spacer size="xs" />);
  expect(container.querySelector('span').className).toBe('spacer spacer--xs');
});

test('displays s spacer', (): void => {
  const { container } = render(<Spacer size="s" />);
  expect(container.querySelector('span').className).toBe('spacer spacer--s');
});

test('displays m spacer', (): void => {
  const { container } = render(<Spacer size="m" />);
  expect(container.querySelector('span').className).toBe('spacer spacer--m');
});

test('displays l spacer', (): void => {
  const { container } = render(<Spacer size="l" />);
  expect(container.querySelector('span').className).toBe('spacer spacer--l');
});

test('displays xl spacer', (): void => {
  const { container } = render(<Spacer size="xl" />);
  expect(container.querySelector('span').className).toBe('spacer spacer--xl');
});

test('displays 2xl spacer', (): void => {
  const { container } = render(<Spacer size="2xl" />);
  expect(container.querySelector('span').className).toBe('spacer spacer--2xl');
});

test('displays 3xl spacer', (): void => {
  const { container } = render(<Spacer size="3xl" />);
  expect(container.querySelector('span').className).toBe('spacer spacer--3xl');
});

test('displays 4xl spacer', (): void => {
  const { container } = render(<Spacer size="4xl" />);
  expect(container.querySelector('span').className).toBe('spacer spacer--4xl');
});

test('displays 5xl spacer', (): void => {
  const { container } = render(<Spacer size="5xl" />);
  expect(container.querySelector('span').className).toBe('spacer spacer--5xl');
});

test('displays 6xl spacer', (): void => {
  const { container } = render(<Spacer size="6xl" />);
  expect(container.querySelector('span').className).toBe('spacer spacer--6xl');
});
