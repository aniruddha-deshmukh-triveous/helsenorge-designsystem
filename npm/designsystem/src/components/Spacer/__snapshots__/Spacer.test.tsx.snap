// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays default spacer 1`] = `
<div>
  <span
    class="spacer spacer--s"
    data-analyticsid="spacer"
  />
</div>
`;
